pkg/webhook/server/webhook_cert_init.go

```diff
@@ -0,0 +1,177 @@
+	if err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+func createSecureService() *v1.Service {
+	label := map[string]string{"app": "che-workspace-controller"}
+
+	port := int32(443)
+	service := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "workspace-controller",
+			Namespace: "che-workspace-controller",
+			Labels:    label,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/serving-cert-secret-name": "workspace-controller",
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Port:       port,
+					Protocol:   "TCP",
+					TargetPort: intstr.FromString("webhook-server"),
+				},
+			},
+			Selector: label,
+		},
+	}
+
+	return service
+}
+
+func createSecureConfigMap() *v1.ConfigMap {
+	return &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "che-workspace-controller-secure-service",
+			Namespace: "che-workspace-controller",
+			Annotations: map[string]string{
+				"service.beta.openshift.io/inject-cabundle": "true",
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+	}
+}
+
+// Update the deployment with the volumes needed for webhook server if they aren't already present
+func updateDeployment(ctx context.Context, crclient client.Client) error {
+
+	log.Info("Attempting to update the deployment with volumes if they are missing")
+	deployment, err := controller.FindControllerDeployment(ctx, crclient)
+	if err != nil {
+		return err
+	}
+
+	isVolumeMissing := appendVolumeIfMissing(&deployment.Spec.Template.Spec.Volumes,
+		v1.Volume{
+			Name: "webhook-tls-certs",
+			VolumeSource: v1.VolumeSource{
+				Projected: &v1.ProjectedVolumeSource{
+					Sources: []v1.VolumeProjection{
+						{
+							ConfigMap: &v1.ConfigMapProjection{
+								LocalObjectReference: v1.LocalObjectReference{
+									Name: "che-workspace-controller-secure-service",
+								},
+								Items: []v1.KeyToPath{
+									{
+										Key:  "service-ca.crt",
+										Path: "./ca.crt",
+									},
+								},
+							},
+						},
+						{
+							Secret: &v1.SecretProjection{
+								LocalObjectReference: v1.LocalObjectReference{
+									Name: "workspace-controller",
+								},
+							},
+						},
+					},
+				},
+			},
+		})
+
+	isVMMissing := appendVolumeMountIfMissing(&deployment.Spec.Template.Spec.Containers[0].VolumeMounts,
+		*&v1.VolumeMount{
+			Name:      "webhook-tls-certs",
+			MountPath: webhookServerCertDir,
+			ReadOnly:  true,
+		})
+
+	// Only bother updating if the volume or volume mount are missing
+	if isVolumeMissing || isVMMissing {
+		log.Info("Attempting to update the deployment with correct volume and volume mounts for secrets")
+		if err = crclient.Update(ctx, deployment); err != nil {
+			log.Info("There was an error")
+			log.Info(err.Error())
+			return err
+		}
+	}
+
+	return nil
+}
+
+// append the volume mount if it is missing. Indicates if the volume mount is missing
```

Comment:
godoc generally suggests starting function docs with the function name:
```suggestion
// appendVolumeMountIfMissing appends the volume mount if it is missing. Indicates if the volume mount is missing
```